### PR TITLE
Add episode 28

### PR DESCRIPTION
**Pull request by bitsondatadev**

### CODE REVIEW

File: _episodes/28.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+ 
+## Release 363
+
+Official announcement items from Martin:
+
+* 
+
+Manfred's additional notes:
+
+ *
```
Comment (mosabua) [inline _episodes/28.md:49]:
```
* Misc performance and memory usage improvements
* `SHOW ROLES` fix
* `EXPLAIN ANALYZE` fix for estimate display
*  Numerous improvements for Parquet files in Hive and Iceberg connectors
```

File: _episodes/28.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+answer is less clear when discussing event-driven systems though. An **event**
+is an action or occurrence that is captured by either a sensor, or a generated
+by a source system, and emitted to a sink system. Some examples include user
+events from an application, system events in telemetry systems, or sensor events
+from monitoring applications.
+
+What is an event stream? Now knowing what an event is, an **event stream** is, an 
+unbounded set of events that are tracked and retained with some notion of time.
+
+The simple view of an event stream will contain a sequential list of events. The
```
Comment (mosabua) [inline _episodes/28.md:77]: In this simple view an event stream contains a sequential list of events.

File: _episodes/28.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+representing event streams.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/event-stream-realistic.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+### Pravega and segments
+
+Engineers at Dell labs wanted to find a better abstraction to solve for the
```
Comment (mosabua) [inline _episodes/28.md:100]: is it Dell Labs ?

File: _episodes/28.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/event-stream-realistic.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+### Pravega and segments
+
+Engineers at Dell labs wanted to find a better abstraction to solve for the 
+problems they saw in existing event streaming systems. This included how to 
+address this type of constant shift in scaling while also addressing the brittle
```
Comment (mosabua) [inline _episodes/28.md:102]: scaling, while

File: _episodes/28.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+storage abstractions that even streams use today. This storage abstraction needs
```
Comment (mosabua) [inline _episodes/28.md:103]: The storage abstraction

File: _episodes/28.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+to allow for both realtime and historical analytics. The data along a particular
+transaction also needs to be consistent.
+
+<p align="center">
+<img align="center" width="50%" height="100%" src="/assets/episode/28/segment.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+
+Their solution is Pravega and the core of Pravega build around modeling streams
```
Comment (mosabua) [inline _episodes/28.md:113]: Pravega. The core of Pravega models streams built around a storage unit. called a segment.

File: _episodes/28.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+built around a storage unit called a segment. A **segment** is an append only
+sequence of bytes (not events/records). This offers a greater level of
```
Comment (mosabua) [inline _episodes/28.md:115]: Change sentence to avoid "offer" twice.

File: _episodes/28.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+sequence of bytes (not events/records). This offers a greater level of 
+flexibility to offer better parallelism and serialization over streams. Pravega
+stream writers are then able to write in parallel increasing ingest throughput.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/autoscale-parallel-segment.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+You can use **routing keys** to map events to particular segments. Pravega will
```
Comment (mosabua) [inline _episodes/28.md:124]: Pravega enforces order

File: _episodes/28.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+enforce order within specific keys but does not guarantee ordering of events
```
Comment (mosabua) [inline _episodes/28.md:125]: keys, but does

File: _episodes/28.md [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+this enables is managing transactions in the stream. As writers submit data,
+they write to a temporary segment, which are merged to a permanent segment on
+commit.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/segment-transactions.png"/><br/>
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+Below shows a very nice way of visualizing autoscaling splits and merges as
```
Comment (mosabua) [inline _episodes/28.md:140]: The following diagram displays autoscaling splits

File: _episodes/28.md [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+Below shows a very nice way of visualizing autoscaling splits and merges as 
+specific routing keys become more popular. To provide a clearer example, say
+that the routing keys are actually just hash geo location values for a taxi app
+that are mapped between zero and one. As certain locations become crowded, lets
+say that a lot of people are going home for the work day, and many taxis are in
+the downtown location. The locations mapped to the downtown routing keys can
+automatically trigger a split, and once the rush hour is over, it will merge
```
Comment (mosabua) [inline _episodes/28.md:146]: it merges these

File: _episodes/28.md [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+these segments as traffic slows down. 
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/segment-split-merge.png"/><br/>
+Pravega Docs: <a href="https://pravega.io/docs/nightly/pravega-concepts/#elastic-streams-auto-scaling">Source</a>.
+</p>
+
+### Pravega Architecture
```
Comment (mosabua) [inline _episodes/28.md:154]: architecture

File: _episodes/28.md [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+Cloud Native Computing Foundation Presentation: <a href="https://www.cncf.io/wp-content/uploads/2020/08/pravega-overview-cncf-apr-2020.pdf">Source</a>.
+</p>
+
+### Pravega and segments
+
+Engineers at Dell labs wanted to find a better abstraction to solve for the 
+problems they saw in existing event streaming systems. This included how to 
+address this type of constant shift in scaling while also addressing the brittle
+storage abstractions that even streams use today. This storage abstraction needs 
+to allow for both realtime and historical analytics. The data along a particular
```
Comment (mosabua) [inline _episodes/28.md:104]: real-time

File: _episodes/28.md [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+these segments as traffic slows down. 
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/segment-split-merge.png"/><br/>
+Pravega Docs: <a href="https://pravega.io/docs/nightly/pravega-concepts/#elastic-streams-auto-scaling">Source</a>.
+</p>
+
+### Pravega Architecture
+
+The Pravega Architecture comes with writers groups and reader groups that scale
```
Comment (mosabua) [inline _episodes/28.md:156]: architecture

File: _episodes/28.md [hunk 15]
```diff
@@ -0,0 +1,257 @@
...
+a controller that maintains stream metadata and the segment store that works off
+of tier one storage (Apache Bookkeeper) and tier two storage (Object storage).
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/28/pravega-architecture.png"/><br/>
+Pravega Docs: <a href="https://pravega.io/docs/nightly/pravega-concepts/#elastic-streams-auto-scaling">Source</a>.
+</p>
+
+Just like Trino, Pravega also aims to build a rich set of connectors with
+systems that will be the source and sink. This includes a connector used for
```
Comment (mosabua) [inline _episodes/28.md:167]: that act as source or sink.

File: _episodes/28.md [hunk 16]
```diff
@@ -0,0 +1,257 @@
...
+| Ingestion of large data (video)    | ✅ |    |    |
+| efficient at high partition counts | ✅ |    |    |
+| Consistent state replication       | ✅ |    |    |
+| Key-value tables                   | ✅ |    |    |
+
+Comparison between Pravega, Kafka, and Pulsar: [Source](https://pravega.io)
+
+## Demo of the week: Querying Pravega from Trino
+
+This week, Pravega will be showing us a demonstration from their own [getting-started](https://github.com/pravega/presto-connector/tree/main/getting-started)
```
Comment (mosabua) [inline _episodes/28.md:197]: This week the Pravega teams demonstrates an example from their getting started tutorial for the Trino connector.

File: _episodes/28.md [hunk 17]
```diff
@@ -0,0 +1,257 @@
...
+This weeks [PR of the week](https://github.com/pravega/presto-connector/pull/49)
+doesn't come from the Trino repository this week but rather the presto-connector
+repository. The Trino portion of the repository was committed by Dell engineer 
+Karan Singh. As it states, this now makes Pravega available from Trino along 
+with the original Presto connector.
+            
+Thanks Karan for adding Trino and Andrew for writing the original Presto-Pravega
+connector!
+
+## Question of the week: What is the point of Trino Forum and what is the relationship to Trino Slack?
```
Comment (mosabua) [inline _episodes/28.md:211]: forum.  ??

File: _episodes/28.md [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+
+Our [question of the week](https://www.trinoforum.org/t/what-is-the-point-of-this-forum-and-what-is-the-relationship-to-trino-slack/28)
+comes from the new Trino Forum by Starburst that Brian and a few others at  have put
```
Comment (mosabua) [inline _episodes/28.md:214]: hm .. maybe you are using it as name and hence always use "Trino Forum" .. up to you

Comment (mosabua) [reply _episodes/28.md:214]: by Starburst. Brian and a few other helpers have set this up and started to add content.

File: _episodes/28.md [hunk 19]
```diff
@@ -0,0 +1,257 @@
...
+Thanks Karan for adding Trino and Andrew for writing the original Presto-Pravega
+connector!
+
+## Question of the week: What is the point of Trino Forum and what is the relationship to Trino Slack?
+
+Our [question of the week](https://www.trinoforum.org/t/what-is-the-point-of-this-forum-and-what-is-the-relationship-to-trino-slack/28)
+comes from the new Trino Forum by Starburst that Brian and a few others at  have put 
+Starburst created. Slack is a much more adhoc platform for people to work 
+through problems rather than to search and find solutions to problems. The Trino
+Community has such a great amount of knowledge accumulated in this Slack channel,
```
Comment (mosabua) [inline _episodes/28.md:217]: community

File: _episodes/28.md [hunk 20]
```diff
@@ -0,0 +1,257 @@
...
+but there is no way for people to find answers unless they have joined here and 
+none of the information we discuss can be found by a search engine like Google. 
+
+Further, a lot of the answers are scattered between different conversations and 
+this too can be condensed and simplified. I pondered about the best way for us 
+to expose this and though maybe to add an FAQ page on <trino.io> but this would
+get stale quickly and this would require a lot of work to be maintained at scale
+without a crowdsourcing element. Instead, starting a [Discourse forum](https)
```
Comment (mosabua) [inline _episodes/28.md:225]: link

File: _episodes/28.md [hunk 21]
```diff
@@ -0,0 +1,257 @@
...
+Community has such a great amount of knowledge accumulated in this Slack channel,
+but there is no way for people to find answers unless they have joined here and 
+none of the information we discuss can be found by a search engine like Google. 
+
+Further, a lot of the answers are scattered between different conversations and 
+this too can be condensed and simplified. I pondered about the best way for us 
+to expose this and though maybe to add an FAQ page on <trino.io> but this would
+get stale quickly and this would require a lot of work to be maintained at scale
+without a crowdsourcing element. Instead, starting a [Discourse forum](https) 
+(not to be confused with Discord) that will act as a central repository of
```
Comment (mosabua) [inline _episodes/28.md:226]: acts as

File: _episodes/28.md (hunk 20 above)
Comment (mosabua) [inline _episodes/28.md:225]: Instead ...  this whole following sections needs to be changed to be a sentence

File: _episodes/28.md [hunk 22]
```diff
@@ -0,0 +1,257 @@
...
+without a crowdsourcing element. Instead, starting a [Discourse forum](https) 
+(not to be confused with Discord) that will act as a central repository of 
+knowledge that is easily searchable by search engines and will be maintained 
+initially by some of us at Starburst but over time more and more from the 
+community (this happens through merit and consistency using Discourse Trust 
+levels). 
+
+## Events, news, and various links
+
+Blogs and Resources
```
Comment (mosabua) [inline _episodes/28.md:234]: resources

**Review by mosabua — CHANGES_REQUESTED** [review]

File: _episodes/28.md [hunk 23]
```diff
@@ -0,0 +1,257 @@
...
+     desc: "Event Stream abstractions and Pravega"
+     time: 915
+   - title: "Demo of the week"
+     desc: "Event Stream abstractions and Pravega"
+     time: 4260
+   - title: "PR of the week"
+     desc: "Pravega presto-connector PR 49"
+     time: 4851
+   - title: "Question of the week"
+     desc: "What is the point of Trino Forum and what is the relationship to Trino Slack?"
```
[the remark this answers is not in the document]
Comment (bitsondatadev) [reply _episodes/28.md:19]: Trino Forum is part of the title so I'll stick with, "Forum, and"

File: _episodes/28.md (hunk 17 above)
Comment (bitsondatadev) [reply _episodes/28.md:211]: should be upper

**Review by mosabua — APPROVED** [review]
Good enough now imho. Let's ship it.

**Comment by bitsondatadev** [comment]
Hey @martint, this one too 😄.

**Pull request merged**